src/routes/taskInfo.js

```
@@ -0,0 +1,90 @@
+const taskRoutes = require("express").Router();
+const tasks = require("../task.json");
+const bodyParser = require("body-parser");
+const path = require("path");
+const fs = require("fs");
+const validator = require("../helper/validator");
+
+taskRoutes.use(bodyParser.urlencoded({ extended: false }));
+taskRoutes.use(bodyParser.json());
+
+//To get all task details
+taskRoutes.get("/", (req, res) => {
+  res.status(200).send(tasks);
+});
+
+//To get task details by task id
+taskRoutes.get("/:id", (req, res) => {
+  let tasklist = tasks.Tasks;
+  let id = req.params.id;
+  let result = tasklist.filter((val) => val.id === id);
+  if (result.length !== 0) {
+    res.status(200).send(result);
+  } else {
+    res.status(404).send(`Task details not available for the id ${id}`);
+  }
+});
+
+//To create a new task
+taskRoutes.post("/", (req, res) => {
+  let taskDetails = req.body;
+  let writepath = path.join(__dirname, "..", "task.json");
+  let taskData = tasks;
+  if (validator.taskInfoValidator(taskDetails).status) {
+    taskData.Tasks.push(taskDetails);
+    fs.writeFileSync(writepath, JSON.stringify(taskData), {
+      encoding: "utf-8",
+      flag: "w",
+    });
+    res.status(200).send("Task is added successfully");
+  } else {
+    res.status(400).send("Task details not added as it is not in the required format");
```

try to follow fail faster approach while writing code, so all the negative cases can be handled in the code first and then you can move ahead with the happy flow.